store/tikv/coprocessor.go

```diff
@@ -545,27 +643,42 @@ func (it *copIterator) open(ctx context.Context) {
 				minCommitTSPushed: &it.minCommitTSPushed,
 				Client:            it.store.client,
 			},
-
-			memTracker: it.memTracker,
-
+			memTracker:      it.memTracker,
 			replicaReadSeed: it.replicaReadSeed,
 		}
 		go worker.run(ctx)
 	}
+	bo := NewBackoffer(ctx, copBuildTaskMaxBackoff).WithVars(it.vars)
+	newTasksCh, err := buildCopTasksChan(bo, it.store.regionCache,
+		&copRanges{mid: it.req.KeyRanges}, it.req)
+	if err != nil {
+		return err
+	}
+
 	taskSender := &copIteratorTaskSender{
-		taskCh:   taskCh,
-		wg:       &it.wg,
-		tasks:    it.tasks,
-		finishCh: it.finishCh,
-		sendRate: it.sendRate,
+		iterator:   it,
+		newTasksCh: newTasksCh,
+		taskCh:     taskCh,
+		wg:         &it.wg,
+		finishCh:   it.finishCh,
+		sendRate:   it.sendRate,
+		respChan:   it.respChan,
 	}
-	taskSender.respChan = it.respChan
 	go taskSender.run()
+	return nil
 }
 
 func (sender *copIteratorTaskSender) run() {
+	// Fill iterator's list of tasks
+	for t := range sender.newTasksCh {
+		sender.iterator.tasks = append(sender.iterator.tasks, t)
+	}
+	// Notify iterator about it
+	sender.iterator.tasksFilledCh <- struct{}{}
+	close(sender.iterator.tasksFilledCh)
```

Comment:
I think we should not wait all tasks done here, otherwise it acts like what it did before.
As soon as a task is built, it should be sent into `sender.taskCh`